src/vs/code/node/cliVerboseLogger.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ChildProcess } from 'child_process';
+import { watch, writeFileSync, unlinkSync } from 'fs';
+import { Promises } from 'vs/base/node/pfs';
+
+/**
+ * This class provides logging for verbose commands on Mac
+ * as part of #102975
+ */
+export class CliVerboseLogger {
+	constructor(readonly filename: string) {
+		writeFileSync(filename, '');
+	}
+
+	public track(): (child: ChildProcess) => Promise<void> {
```

Comment:
I find this a bit weird to pass in the `ChildProcess` from the outside only to finish the watcher on `close`. I would do this from the outside.

Reply:
I'm not too sure what you mean here. Pushing up a small refactor.

Reply:
I mean to build a promise such as `Promise.any(one, two)` from the outside, where `one` is the child process on close event.